include/zephyr/device.h

```diff
@@ -1080,11 +1080,22 @@ device_get_dt_nodelabels(const struct device *dev)
 			(.pm_constraints = (constraints_),))				\
 		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,				\
 			(.pm_constraints_size = (constraints_size_),))			\
-		IF_ENABLED(CONFIG_PM_DEVICE, ({ .pm_base = (pm_),})) /**/		\
+		IF_ENABLED(CONFIG_PM_DEVICE, Z_DEVICE_INIT_PM_BASE(pm_)) /**/		\
 		IF_ENABLED(CONFIG_DEVICE_DT_METADATA,					\
 			   (.dt_meta = &Z_DEVICE_DT_METADATA_NAME_GET(dev_id_),))	\
 	}
 
+/*
+ * Anonymous unions require C11. Some pre-C11 gcc versions have early support for anonymous
+ * unions but they require these braces when combined with C99 designated initializers. For
+ * more details see https://docs.zephyrproject.org/latest/develop/languages/cpp/
+ */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__) < 201100
+#  define Z_DEVICE_INIT_PM_BASE(pm_) ({ .pm_base = (pm_),})
```

Comment:
isn't this missing a trailing `,`?

Reply:
see https://github.com/zephyrproject-rtos/zephyr/pull/74970

Reply:
Great catch, thank you @gmarull for the fix in #74970. Note this regression is only 3 weeks old and independent of this PR. My bad though for not noticing it when I rebased on top of it.

So I just:
- rebased on top of your comma fix #74970 
- Added `CONFIG_DEVICE_DT_METADATA` to the `prj.conf`, on top of the other additions that were already in this PR. I checked it would have caught this bug.